lib/rules/id-match.js

```diff
@@ -67,6 +67,8 @@ module.exports = {
             onlyDeclarations = !!options.onlyDeclarations,
             ignoreDestructuring = !!options.ignoreDestructuring;
 
+        const globalScope = context.getScope();
```

Comment:
Can we wrap this in `Program() {}`?

https://github.com/eslint/eslint/blob/3928175d01c6ac2b37147b3256c56df8faf2c6c4/lib/rules/id-denylist.js#L213-L215

Formally, `context.getScope()` returns the scope of the currently-traversed node, but at this point traversal hasn't started yet. Technically, this works because linter sets [`currentNode = sourceCode.ast`](https://github.com/eslint/eslint/blob/3928175d01c6ac2b37147b3256c56df8faf2c6c4/lib/linter/linter.js#L960) before calling `create()` of rules, but I think that isn't guaranteed. I just checked all `context.getScope()` in core rules, and we're never calling it directly in `create()`.

Reply:
Makes sense, updated 👍🏻